packages/fxa-auth-server/lib/payments/configuration/base.ts

```diff
@@ -36,6 +40,18 @@ const urlsSchema = joi.object({
   appStore: joi.string().uri(),
 });
 
+export const UiContentConfigKeys = [
+  'subtitle',
+  'details',
+  'successActionButtonLabel',
+  'upgradeCTA',
+] as const;
+type UiContentConfigKeysT = typeof UiContentConfigKeys;
+// TODO: How to get `details` as an array of strings? Figure out and swap
+// types used below; should `details` have a sub-subtype? Would that even help?
+// export type UiContentConfig = {
+//   [key in UiContentConfigKeysT[number]]?: string;
+// };
```

Comment:
If the config keys were defined using a dictionary instead of an array, then you can do something like 
```
export type UiContentConfig = {
  [k in keyof Omit<UiContentConfigKeysT, 'details']?: string;
} & { details?: string[] };
```

I don't know a way to exclude an element from a const array type.

Reply:
Thank you Barry that worked beautifully!

Reply:
You can remove the const assertion from the UiContentConfigKeys object.

Reply:
Thanks for the suggestion; that does prevent the read only error, but then I don't have the desired constraint on the type of `key`; i.e. it's no longer a sum type.

With the const assertion, in `uiContentMetadataToUiContentConfig`, it is correctly `const key: "subtitle" | "successActionButtonLabel" | "upgradeCTA"`, whereas without it, `const key: string`, and I get a different TS error, since the return value of the method should be of type `UiContentConfig`:
```
Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'UiContentConfig'.
  No index signature with a parameter of type 'string' was found on type 'UiContentConfig'.
```

I'll think about it a little more.